### PR TITLE
Add confidence to geocodes & added geocodio module

### PR DESCRIPTION
**Pull request by mattkrick**
Just because I get a result doesn't mean it's right. For example, I may try to geocode a house address & only get the lat lng of a city. Ideally, if a service isn't accurate enough, I could retry it with another service (that's why your package is great).

That's what `confidence` is for. Within the extras object (for google, openmapquest, opencage, and geocodio) there's a confidence score from 0 to 1 (This is subjectively normalized based on my experience with each, although I tried to document as much as I could). A 1 means rooftop, .9 means interpolated, .8 means street, .7 means neighborhood-ish, etc. I've never used the other services, but could add confidence to those as well if desired.

Additionally, I added a module for geocod.io, which is a great source for TIGER data in the US.

### CODE REVIEW

**Comment by nchaulet** [comment]
Great :+1:

**Comment by nchaulet** [comment]
Could you fix the test and the lint typos : using `npm test` and `npm run lint` ?

**Comment by mattkrick** [comment]
whoops, my bad, first time playing with mocha...
Should be all good

File: lib/geocoder/googlegeocoder.js [hunk 1]
```diff
@@ -28,222 +28,205 @@ util.inherits(GoogleGeocoder, AbstractGeocoder);
...
-        }
+GoogleGeocoder.prototype._formatResult = function (result) {
+
+  var googleConfidenceLookup = {
+    ROOFTOP: 1,
+    RANGE_INTERPOLATED: 0.9,
+    GEOMETRIC_CENTER: 0.7,
+    APPROXIMATE: 0.5
+  };
+  console.log(googleConfidenceLookup[result.geometry.location_type]);
```
Comment (nchaulet) [inline lib/geocoder/googlegeocoder.js]: could you remove the console.log ?

**Pull request merged**